src/ValidatorKeys.h

```diff
@@ -0,0 +1,132 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of validator-keys-tool:
+        https://github.com/ripple/validator-keys-tool
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/crypto/KeyType.h>
+#include <ripple/protocol/SecretKey.h>
+#include <boost/filesystem.hpp>
+
+namespace ripple {
+
+struct EphemeralKeys
+{
+    Seed const seed;
+    std::string const manifest;
+    PublicKey const publicKey;
+};
+
+class ValidatorKeys
+{
+private:
+    KeyType keyType_;
+    PublicKey masterPublicKey_;
+    SecretKey masterSecret_;
+    std::uint32_t sequence_;
+
+public:
+    explicit
+    ValidatorKeys (
+        KeyType const& keyType);
+
+    ValidatorKeys (
+        KeyType const& keyType,
+        PublicKey const& masterPublicKey,
+        SecretKey const& masterSecret,
+        std::uint32_t sequence);
+
+    /** Returns ValidatorKeys constructed from JSON file
+
+        @param keyFile Path to JSON key file
+
+        @throws std::runtime_error if file content is invalid
+    */
+    static ValidatorKeys make_ValidatorKeys(
+        boost::filesystem::path const& keyFile);
+
+    ~ValidatorKeys () = default;
+
+    inline bool
+    operator==(ValidatorKeys const& rhs) const
+    {
+        return sequence_ == rhs.sequence_ && keyType_ == rhs.keyType_ &&
+               masterPublicKey_ == rhs.masterPublicKey_;
```

Comment:
I think encoding is a heck of a lot of overhead for a comparison that's only used in tests.

I suggest add a comparison operator for `SecretKey` to rippled. It looks like it could be identical to the comparison for `PublicKey`. Update this function once it's merged. (No rush.)